server/test/sc/server/helpers/ConverterTest.kt

```diff
@@ -0,0 +1,49 @@
+package sc.server.helpers
+
+import org.junit.Assert
+import org.junit.Test
+import sc.framework.plugins.IPerspectiveAware
+import sc.framework.plugins.IPerspectiveProvider
+import sc.helpers.xStream
+
+class ConverterTest {
+    class HasSecrets : IPerspectiveProvider, IPerspectiveAware {
+        val secret = "i-am-secret"
+        val unimportant = "i-am-unimportant"
+        private var perspective: Any? = null
+        override fun getPerspective(): Any {
+            return perspective!!
+        }
+
+        fun setPerspective(o: Any?) {
+            perspective = o
+        }
```

Comment:
I can't remove getter and setter because getPerspective overrides an interface method, while setPerspective doesn't.
Plus, as far as I see it, perspective doesn't need a more specific type because nothing is done with the type.

Reply:
we should get rid of the `!!`, that's why I mentioned to check the types. As for the overriding, if we could get the field non-nullable that shouldn't be an issue...

Reply:
ed1e1c19 &
fda81a06

Reply:
Thanks, that looks much better :)